### DISCUSSION TITLE
Should I use ObservableChangeSet.Create or Observable.StartAsync to load data from a task if the data is not going to change?

### DISCUSSION THREAD

**Discussion by Snailya** [reactivemarbles/DynamicData#953] [Q&A]
In dynamicdata.snippets, there is a demo code for ObservableChangeSet.Create() which loads data from a task. I think these code could be for loading the initial data in the viewmodel. However, I don't know whether these could is for demostration only, because as the data is never going to be change, why should I use ObservableChangeSet.Create other than use Observable.StartAsync?
```
        /// <summary>
        /// Create an observable change set from a task
        /// </summary>
        /// <returns></returns>
        public static IObservable<IChangeSet<int>> FromTask()
        {
            return ObservableChangeSet.Create<int>(async list =>
            {
                var items = await LoadFromTask();
                list.AddRange(items);
                return () => { };
            });
        }

        public static IObservable<IChangeSet<int>> FromTask(Func<Task<IEnumerable<int>>> loader)
        {
            return ObservableChangeSet.Create<int>(async list =>
            {
                var items = await loader();
                list.AddRange(items);
                return () => { };
            });
        }
```

**Comment by JakenVeina** [comment]
> why should I use ObservableChangeSet.Create other than use Observable.StartAsync?

Because (presumably) your `Task` isn't returning a changeset, it's returning a collection of items. `Observable.StartAsync()` doesn't create a DynamicData collection stream that you can invoke DynamicData operators on.
